### ISSUE TITLE
freeze layers weight

### ISSUE THREAD

**Issue by gdkrmr** [dmlc/MXNet.jl#63]
Is there a way to freeze layer weights?

**Comment by pluskid** [comment]
Currently there is no easy way of doing it. One workaround is to create a new network and insert a [BlockGrad](http://mxnetjl.readthedocs.org/en/latest/api/symbolic-node.html#BlockGrad) node above the part that you want to freeze.

**Comment by gdkrmr** [comment]
Wouldn't that freeze anything below the `BlockGrad` node?

**Comment by pluskid** [comment]
@gdkrmr Yes, that's a drawback for that workaround. In some cases (e.g. bottom up layerwise pre-training) that is actually the expected behavior. Another workaround, which might get quite complicated, is to modify the existing model.jl code, and use a different learning rate (e.g. 0) for different parameters.

**Comment by gdkrmr** [comment]
ok, thanks! I will try to figure it out, you are right, setting per layer independent learning rates would be quite useful.

**Comment by vchuravy** [comment]
I am currently looking into this for fine-tuning and transfer learning. The problem with the `BlockGrad` approach is that weight_decay will still happen. 

I was looking into making and attribute `:grad => "freeze"` work, but I was stumbling over how we update the gradients.

Places we would need to adapt are:
https://github.com/dmlc/MXNet.jl/blob/master/src/model.jl#L479-L486,

but I am not quite understanding how to adapt the KVStore case.

Am I right in thinking that in order to freeze the weights, we just don't want to call the updater on them?
